### ISSUE TITLE
Map not refreshed after updating

### ISSUE THREAD

**Issue by gonzaloriestra** [CartoDB/cartodb#15609]
### Context
https://groups.google.com/forum/#!msg/cartodb/A6BacqchvYw/36QHxLMmAgAJ

### Steps to Reproduce
https://www.youtube.com/watch?v=vyk-5MPxAns

1. Add a geometry on a published map
2. Go to the map view and try several zoom levels

### Current Result
The new geometry does not always appear

### Expected result
It should appear for any zoom level

@CartoDB/rt-managers

**Comment by ztephm** [comment]
possibly related? https://github.com/CartoDB/support/issues/2359

**Comment by oleurud** [comment]
Yes, we have several issues about that, but this is a by design "error". I think in the cartodb group Santana explained it very well, let me show you the same with an easy example.

### Create, publish and show a map for first time.
We create and publish a map from builder. When we show it, there are 2 types of request we must know to understand the problem:
- **1. Instantiation**: we need to do it before requesting tiles, to know what is the tiles URL

![image](https://user-images.githubusercontent.com/421145/80716171-9f286000-8af7-11ea-8c5c-f6186adba77a.png)


In the example we have: 
```
"https://cartocdn-gusc-b.global.ssl.fastly.net/simon-carto/api/v1/map/simon-carto@23f4b03a@616654680506b79c1de70e1029a79786:1588252402281/{z}/{x}/{y}.png"
```

-  **2. Tiles**: based on that URL, you can ask for tiles using the zoom, x and y coords. For example:
```
https://cartocdn-gusc-b.global.ssl.fastly.net/simon-carto/api/v1/map/simon-carto@23f4b03a@616654680506b79c1de70e1029a79786:1588252402281/1/3/1/4.png
```

### Update and show the map again.

-  **1. Instantiation** How we have updated our map data, the tiles URLs have been changed:

![image](https://user-images.githubusercontent.com/421145/80716518-06461480-8af8-11ea-9b46-c07584644cc2.png)

```
https://cartocdn-gusc-a.global.ssl.fastly.net/simon-carto/api/v1/map/simon-carto@23f4b03a@616654680506b79c1de70e1029a79786:1588252162538/{z}/{x}/{y}.png
```

 - **2. Tiles**: 

```
https://cartocdn-gusc-a.global.ssl.fastly.net/simon-carto/api/v1/map/simon-carto@23f4b03a@616654680506b79c1de70e1029a79786:1588252162538/1/3/1/4.png
```

### What has changed?

Please, pay attention to the end of the tiles URL: at the end, you can see a number
```
after creation: ...:1588252162538/{z}/{x}/{y}.png
after update :  ...:1588252402281/{z}/{x}/{y}.png
```

That way, we are avoiding cache problems, in our CDN or in Browser since the final URL is different.

### The problem
- you have loaded a map, the first instantiation request is made, and the tiles are requested based on it. 
- the map data is updated
- you are still on the map, so not instantiation is made, so tiles are requested using old timestamp.

### But sometimes it works!
Yes, when the data is uploaded, we send a request to our CDN to flush caches. So in theory, when you ask again for a tile with an old timestamp, the backend will respond with the new data. But, as the URL has not changed, if the browser has cached the tile, it is not going to be requested and the old data will be showed.

### Solution

Instantiate the map again. So in builder: refresh the page

**Comment by dgaubert** [comment]
Hi!

They're editing a map in a tab of their browser and in the other one they have the published map loaded already. If they edit the map in the first tab then the published map (public view) isn't going to be updated automatically, they need to refresh the tab to get the fresh data. 

What they are seeing in the second tab is the published map with mixed content, the stalled one because it's cached and the updated data as long as they are zooming and moving into the map and the tiles weren't previously fetched. That's something that only happens when you update the data and you don't publish the map again and re-load the public view. So the workflow is: edit the map, publish it and (re)load the public view.

That's how the product is designed there is no such thing as Real-Time CARTO (yet). We can mitigate the use case making the public view a bit smarter and detect when the content is stalled and then notify the user to refresh the public view.

HTH

**Comment by knownasilya** [comment]
> we send a request to our CDN to flush caches

Could you point me to where that happens?

**Comment by oleurud** [comment]
> Could you point me to where that happens?

It is not an easy answer :) But I will try to give you some insights.

Everything starts in the database. The database should have [cartodb postgreSQL extension](https://github.com/CartoDB/cartodb-postgresql) installed and the table should have `CDB_CartodbfyTable` function applied (more info https://carto.com/developers/sql-api/guides/creating-tables/). 

`CDB_CartodbfyTable` adds some triggers to the table, one of them is:
```
track_updates AFTER INSERT OR DELETE OR UPDATE OR TRUNCATE ON yourtable FOR EACH STATEMENT EXECUTE PROCEDURE cdb_tablemetadata_trigger()
```
which is the one starting the flushing cache process https://github.com/CartoDB/cartodb-postgresql/blob/master/scripts-available/CDB_TableMetadata.sql#L30-L62. 

After that, a lot of stuff happens but in the end, varnish and CDN are flushed when required.
